src/main/java/de/sakpaas/backend/service/SearchService.java

```diff
@@ -97,6 +101,26 @@ protected SearchRequest dbBrandSearch(SearchRequest request) {
    * @return the updated Request Object
    */
   protected SearchRequest getByCoordinates(SearchRequest request) {
+    // Get Locations
+    CoordinateDetails coordinateDetails = request.getCoordinates();
+    List<Location> locations = locationService
+        .findByCoordinates(coordinateDetails.getLatitude(), coordinateDetails.getLongitude());
+
+    // Filter by brand
+    if (knownBrands.isEmpty()) {
+      request.setLocations(locations);
+    } else {
+      List<Location> locationList = new ArrayList<>();
+      for (String brand : knownBrands) {
+        locationList.addAll(
```

Comment:
Persönlich würde ich die Logik hier umdrehen und die `for`-Schleife in dem `filter()` haben, damit es einen kleinen Ticken effizienter ist, da man sonst so oft `addAll` ausführen muss. Ist aber nur minimal. :)

Reply:
Kannst du mir ein Code-Beispiel geben, weil ich sonst nicht verstehe, was du meinst.